test/unit/org/apache/cassandra/repair/RepairJobTest.java

```diff
@@ -274,9 +277,9 @@ public void testNoTreesRetainedAfterDifference() throws Throwable
         long millisUntilFreed;
         for (millisUntilFreed = 0; millisUntilFreed < TEST_TIMEOUT_S * 1000; millisUntilFreed += THREAD_TIMEOUT_MILLIS)
         {
-            // The measured size of the syncingTasks, and result of the computation should be much smaller
+            // Once the threads die, the size of the session should return to nearly pre-execution levels.
             TimeUnit.MILLISECONDS.sleep(THREAD_TIMEOUT_MILLIS);
-            if (ObjectSizes.measureDeep(session) < 0.8 * singleTreeSize)
+            if (ObjectSizes.measureDeep(session) <= (sizeBeforeTasks * 1.05))
```

Comment:
The starting size of the session (before a single task is submitted) is much higher in Java 11 than Java 8 for some reason, but the amount of memory freed by the dying threads is almost exactly the same. The idea here is that we simply remember the starting size and make sure we revert to nearly that once the workers expire.

Reply:
to help maintenance it would be good to document this

Reply:
>  revert to nearly that once the workers expire.
what is the extra 5% for?

Reply:
given how I read the test, this patch looks fine to me; just making sure we revert to where we were before starting; which is mostly what this test is checking....

+1 for that, but would still be nice to know what got larger.

Reply:
I'll make a note around this in the Jira and expand the comment inline here as well.